test/unit/trace-test.js

```diff
@@ -247,14 +248,405 @@ test('test trace serialization, three spans', (t) => {
+                uri: null,
+                timestamp: childSpan2.getStartTime() * 1000,
+                operation: childSpan2.getOperationName(),
+                duration: childSpan2.getDuration() * 1000,
+                nodes: [{
+                    type: NODE_TYPE_COMPONENT,
+                    uri: null,
+                    timestamp: childSpan22.getStartTime() * 1000,
+                    operation: childSpan22.getOperationName(),
+                    duration: childSpan22.getDuration() * 1000,
+                    nodes:[{
+                        type: NODE_TYPE_COMPONENT,
+                        uri: null,
+                        timestamp: multipleReferencesSpan.getStartTime() * 1000,
+                        operation: multipleReferencesSpan.getOperationName(),
+                        duration: multipleReferencesSpan.getDuration() * 1000,
+                        correlationIds: [{
+                            value: `${rootSpan.getId()}:0:0:0`,
+                            scope: CORR_ID_SCOPE_CAUSED_BY
+                        }],
+                    }],
+                }],
+            }],
+        }],
+    });
+    t.end();
+});
+
+test('test mixed references, primary parent extracted context', (t) => {
+    const recorder = new ListRecorder();
+    const tracer = new APMTracer({ recorder: recorder });
+
+    const extractedContext = tracer.extract(Constants.FORMAT_TEXT_MAP, {});
+    const rootSpan = tracer.startSpan('root', { childOf: extractedContext});
+    const childSpan1 = tracer.startSpan('child1', { childOf: rootSpan });
+    const childSpan11 = tracer.startSpan('child11', { childOf: childSpan1 });
+    const childSpan2 = tracer.startSpan('child2', { childOf: rootSpan });
+    const childSpan22 = tracer.startSpan('child22', { childOf: childSpan2 });
+
+    childSpan1.finish();
+    childSpan11.finish();
+    childSpan2.finish();
+    rootSpan.finish();
+
+    const multipleReferencesSpan = tracer.startSpan('follows', {
+        references: [
+            new Constants.Reference(Constants.REFERENCE_FOLLOWS_FROM, extractedContext),
+            new Constants.Reference(Constants.REFERENCE_CHILD_OF, childSpan22.context()),
+        ]});
+
+    multipleReferencesSpan.finish();
+    childSpan22.finish();
+
+    t.deepEqual(recorder.getTraces().length, 2);
+    t.deepEquals(recorder.getTraces()[0], {
+        traceId: multipleReferencesSpan.getTraceId(),
+        fragmentId: multipleReferencesSpan.getId(),
+        timestamp: multipleReferencesSpan.getStartTime() * 1000,
+        nodes: [{
+            type: NODE_TYPE_CONSUMER,
+            endpointType: 'HTTP',
+            uri: null,
+            operation: multipleReferencesSpan.getOperationName(),
+            timestamp: multipleReferencesSpan.getStartTime() * 1000,
+            duration: multipleReferencesSpan.getDuration() * 1000,
+            correlationIds: [{
+                value: `${rootSpan.getId()}:0:1:0`,
+                scope: CORR_ID_SCOPE_INTERACTION
```

Comment:
Might be good having this test, with 'empty trace state' extracted context. And then have a second version of the same test with trace state supplied?